main/service/external/external_exchange_api_controller.go

```diff
@@ -0,0 +1,49 @@
+package external
+
+import (
+	"container/list"
+	"errors"
+
+	"github.com/sirupsen/logrus"
+	"ses.genesis.com/exchange-web-service/main/config"
+
+	"github.com/go-resty/resty/v2"
+)
+
+type RateService interface {
+	CurrentRate(from string, to string) (float64, error)
+}
+
+type ExchangeAPIController struct {
+	config       *config.AppConfig
+	client       *resty.Client
+	externalAPIs *list.List
+}
+
+func NewExternalExchangeAPIController(conf *config.AppConfig, client *resty.Client, apis *list.List) *ExchangeAPIController {
+	return &ExchangeAPIController{
+		config:       conf,
+		client:       client,
+		externalAPIs: apis,
+	}
+}
+
+func (controller *ExchangeAPIController) CurrentRate(from, to string) (float64, error) {
+	return getRate(controller.externalAPIs.Front(), from, to)
+}
+
```

Comment:
```suggestion
type RateAPI interface {
	GetRate(from, to string) (float64, error)
}
```
Я б переніс би інтерфейс зі `coinapi_repository.go` саме у місце використання, тобто тут

Reply:
Still unusual to do this. Done.